plugins/pipelines/badge.js

```diff
@@ -43,26 +65,29 @@ module.exports = () => ({
             return factory.get(request.params.id)
                 .then((pipeline) => {
                     if (!pipeline) {
-                        return reply.redirect(getUrl(request.server.app.ecosystem.badges));
+                        return reply.redirect(getUrl(badgeService));
                     }
 
-                    return pipeline.jobs.then((jobs) => {
-                        const main = jobs.filter(job => job.name === 'main').pop();
+                    return pipeline.getEvents({ sort: 'ascending' }).then((events) => {
+                        const lastEvent = events.pop();
 
-                        if (!main) {
+                        if (!lastEvent) {
                             return reply.redirect(getUrl(badgeService));
                         }
 
-                        return main.getBuilds({
-                            paginate: {
-                                page: 1,
-                                count: 1
-                            },
-                            sort: 'descending'
-                        }).then((builds) => {
-                            const build = builds.pop();
+                        return lastEvent.getBuilds().then((builds) => {
+                            if (!builds || builds.length < 1) {
+                                return reply.redirect(getUrl(badgeService));
+                            }
+
+                            const buildsStatus = builds.reverse()
+                                .map(build => build.status.toLowerCase());
+
+                            for (let i = builds.length; i < lastEvent.workflow.length; i += 1) {
```

Comment:
This is not a good longterm approach, workflow is not always in sequential order.  But it will work for now.